### PR TITLE
Migrate FlashCardsContract to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Migrates FlashCardsContract.java to kotlin following the migration guide.

For reviewers:

While the code migration was trivial the documentation failed completely as snippets of code and the tables failed to render. KDoc apparently isn't able to show tables so I had to change the docs to use ``` which keeps the indentation and manually create the tables. I think it looks pretty decent but it has two issues: we can't  reference other classes(with brackets like using [Note] ) and we can't create links inside the triple ` blocks. These two issues can be seen in the table from the documentation for the Note object.

I added two cleanups on the main FlashCardsContract object(in a TODO as we don't have the annotation here), one for the usual ide lint cleanups and one as an improvement for the code in this module with kotlin's explicitapi mode (see https://kotlinlang.org/docs/whatsnew14.html#explicit-api-mode-for-library-authors )

## How Has This Been Tested?

Ran the tests.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: api/src/main/java/com/ichi2/anki/FlashCardsContract.kt [hunk 1]
```diff
@@ -0,0 +1,962 @@
...
+     * String | FLDS   | read-write | Fields of this note. Fields are separated by "\\x1f", a.k.a. Consts.FIELD_SEPARATOR
+     * --------------------------------------------------------------------------------------------------------------------
+     * long   | SFLD   | read-only  | see [main documentation](https://github.com/ankidroid/Anki-Android/wiki/Database-Structure)
+     * --------------------------------------------------------------------------------------------------------------------
+     * long   | CSUM   | read-only  | see [main documentation](https://github.com/ankidroid/Anki-Android/wiki/Database-Structure)
+     * --------------------------------------------------------------------------------------------------------------------
+     * long   | FLAGS  | read-only  | see [main documentation](https://github.com/ankidroid/Anki-Android/wiki/Database-Structure)
+     * --------------------------------------------------------------------------------------------------------------------
+     * long   | DATA   | read-only  | see [main documentation](https://github.com/ankidroid/Anki-Android/wiki/Database-Structure)
```
Comment (david-allison) [inline api/src/main/java/com/ichi2/anki/FlashCardsContract.kt:188]: These links don't work, as they're within code blocks

**Review by david-allison — CHANGES_REQUESTED** [review]
The main documentation looks better now in my opinion.

There's one question about moving the database structure links around, but this looks good

I do worry about the potential API changes, but we are calling this a 'major' version bump, so I'm happy

File: api/src/main/java/com/ichi2/anki/FlashCardsContract.kt (hunk 1 above)
Comment (lukstbit) [reply api/src/main/java/com/ichi2/anki/FlashCardsContract.kt:188]: Yes, I mentioned this in the origin message. What do you have in mind?
I was changing it to:

> _See more at https://github.com/ankidroid/Anki-Android/wiki/Database-Structure_

Comment (david-allison) [reply api/src/main/java/com/ichi2/anki/FlashCardsContract.kt:188]: That works

**Review by david-allison — APPROVED** [review]

**Comment by mikehardy** [comment]
Just tagging @hanpingchinese with another note that the underlying AnkiDroid API should undergo integration testing with your app on next release. We are not changing anything - *we think* - but simply converting it to Kotlin means there may be null/non-null changes that might affect you and other API users

**Review by mikehardy — APPROVED** [review]
Seems fine - thank you for this!

**Pull request merged**